### ISSUE TITLE
AN4, the TrainingCriterion is NAN?

### ISSUE THREAD

**Issue by jren2012** [Microsoft/CNTK#145]
I used the default parameters in the config file "FeedForward.cntk" as: 
**trainingCriterion = "CrossEntropyWithSoftmax"**.
The error described in the title occurred.  How can I fix it? revise the learning rate or some other suggestions?
Any help is appreciated.
@alexeyo26 thanks your answer yesterday
--------------------------------------------------------output ---------------------------------------------------------------
[CALL STACK]
  /home/jren/Repos/cntk/build/release/bin/../lib/libcntkmath.so ( Microsoft::MSR::CNTK::DebugUtil::PrintCallStack()  + 0xc3  )  [0x7faadab7f053]
  cntk                           ( void Microsoft::MSR::CNTK::ThrowFormatted<std::runtime_error>(char const*, ...)  + 0xdd  )  [0x53dccd]
  cntk                           ( Microsoft::MSR::CNTK::SGD<float>::TrainOneEpoch(std::shared_ptr<Microsoft::MSR::CNTK::ComputationNetwork>, std::shared_ptr<Microsoft::MSR::CNTK::ComputationNetwork>, std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase> const&, int, unsigned long, Microsoft::MSR::CNTK::IDataReader<float>*, double, unsigned long, std::vector<std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase>, std::allocator<std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase> > > const&, std::vector<std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase>, std::allocator<std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase> > > const&, std::vector<std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase>, std::allocator<std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase> > > const&, std::vector<std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase>, std::allocator<std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase> > > const&, std::map<std::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t> >, Microsoft::MSR::CNTK::Matrix<float>*, std::less<std::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t> > >, std::allocator<std::pair<std::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t> > const, Microsoft::MSR::CNTK::Matrix<float>*> > >*, std::list<std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase>, std::allocator<std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase> > > const&, std::list<Microsoft::MSR::CNTK::Matrix<float>, std::allocator<Microsoft::MSR::CNTK::Matrix<float> > >&, double&, std::vector<double, std::allocator<double> >&, unsigned long&, std::string)  + 0x2c26)  [0x76e3c6]
  cntk                           ( Microsoft::MSR::CNTK::SGD<float>::TrainOrAdaptModel(int, std::shared_ptr<Microsoft::MSR::CNTK::ComputationNetwork>, bool, std::shared_ptr<Microsoft::MSR::CNTK::ComputationNetwork>, std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase>, Microsoft::MSR::CNTK::IDataReader<float>*, Microsoft::MSR::CNTK::IDataReader<float>*)  + 0x133b)  [0x771eeb]
  cntk                           ( Microsoft::MSR::CNTK::SGD<float>::Train(std::function<std::shared_ptr<Microsoft::MSR::CNTK::ComputationNetwork> (int)>, int, Microsoft::MSR::CNTK::IDataReader<float>*, Microsoft::MSR::CNTK::IDataReader<float>*, bool)  + 0x441 )  [0x774831]
  cntk                           ( void DoTrain<Microsoft::MSR::CNTK::ConfigParameters, float>(Microsoft::MSR::CNTK::ConfigParameters const&)  + 0x22d )  [0x7906ad]
  cntk                           ( void DoCommands<float>(Microsoft::MSR::CNTK::ConfigParameters const&)  + 0x7bc )  [0x5a64ac]
  cntk                           ( wmainOldCNTKConfig(int, wchar_t**)        + 0xacb )  [0x53426b]
  cntk                           ( wmain1(int, wchar_t**)                    + 0x6a  )  [0x534a1a]
  cntk                           ( main                                      + 0xcc  )  [0x526e1c]
  /lib/x86_64-linux-gnu/libc.so.6 ( __libc_start_main                         + 0xf0  )  [0x7faad97baa40]
  cntk                           ( _start                                    + 0x29  )  [0x52aab9]
EXCEPTION occurred: The training criterion is not a number (NAN). Stop

**Comment by dongyu888** [comment]
EXCEPTION occurred: The training criterion is not a number (NAN). Stop

 

Means the training diverged. Either model has probem or the hyper parameters have problem.

**Comment by jren2012** [comment]
Thanks @dongyu888 
I ran the command again, it outputted normal results.
